internal/resource/resource_plugin.go

```diff
@@ -188,53 +188,165 @@ func (r *Resource) handleNginxPlusActionRequest(ctx context.Context, action *mpi
+		r.handleGetUpstreams(ctx, action, instance)
+	default:
+		slog.DebugContext(ctx, "NGINX Plus action not implemented yet")
+	}
+}
 
-			return
-		}
+// nolint: dupl
+func (r *Resource) handleUpdateStreamServers(ctx context.Context, action *mpi.NGINXPlusAction, instance *mpi.Instance) {
+	correlationID := logger.GetCorrelationID(ctx)
+	instanceID := instance.GetInstanceMeta().GetInstanceId()
 
-		slog.DebugContext(ctx, "Successfully updated http upstream servers", "http_upstream_name",
-			action.GetUpdateHttpUpstreamServers().GetHttpUpstreamName(), "add", len(add), "update", len(update),
-			"delete", len(del))
-		resp := r.createDataPlaneResponse(correlationID, mpi.CommandResponse_COMMAND_STATUS_OK,
-			"Successfully updated HTTP Upstreams", instanceID, "")
+	slog.DebugContext(ctx, "Updating stream servers", "request", action.GetUpdateStreamServers())
 
+	add, update, del, err := r.resourceService.UpdateStreamServers(ctx, instance,
+		action.GetUpdateStreamServers().GetUpstreamStreamName(), action.GetUpdateStreamServers().GetServers())
+	if err != nil {
+		slog.ErrorContext(ctx, "Unable to update stream servers of upstream", "request",
+			action.GetUpdateHttpUpstreamServers(), "error", err)
+		resp := r.createDataPlaneResponse(correlationID, mpi.CommandResponse_COMMAND_STATUS_FAILURE,
+			"", instanceID, err.Error())
 		r.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: resp})
 
-	case *mpi.NGINXPlusAction_GetHttpUpstreamServers:
-		slog.DebugContext(ctx, "Getting http upstream servers", "request", action.GetGetHttpUpstreamServers())
-		upstreams, err := r.resourceService.GetUpstreams(ctx, instance,
-			action.GetGetHttpUpstreamServers().GetHttpUpstreamName())
-		if err != nil {
-			slog.ErrorContext(ctx, "Unable to get HTTP servers of upstream", "error", err)
-			resp := r.createDataPlaneResponse(correlationID, mpi.CommandResponse_COMMAND_STATUS_FAILURE,
-				"", instanceID, err.Error())
-			r.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: resp})
+		return
+	}
 
-			return
-		}
+	slog.DebugContext(ctx, "Successfully updated stream upstream servers", "http_upstream_name",
+		action.GetUpdateHttpUpstreamServers().GetHttpUpstreamName(), "add", len(add), "update", len(update),
+		"delete", len(del))
 
-		upstreamsJSON, err := json.Marshal(upstreams)
-		if err != nil {
-			slog.ErrorContext(ctx, "Unable to marshal http upstreams", "err", err)
-		}
-		resp := r.createDataPlaneResponse(correlationID, mpi.CommandResponse_COMMAND_STATUS_OK,
-			string(upstreamsJSON), instanceID, "")
+	resp := r.createDataPlaneResponse(correlationID, mpi.CommandResponse_COMMAND_STATUS_OK,
+		"Successfully updated stream upstream servers", instanceID, "")
+
+	r.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: resp})
+}
+
+// nolint: dupl
+func (r *Resource) handleGetStreamUpstreams(ctx context.Context, action *mpi.NGINXPlusAction, instance *mpi.Instance) {
```

Comment:
you could probably do the log in the switch and make the functions smaller? 

```	correlationID := logger.GetCorrelationID(ctx)
	instanceID := instance.GetInstanceMeta().GetInstanceId()

	slog.DebugContext(ctx, "Getting stream upstreams", "request", action.GetGetStreamUpstreams())```

Reply:
also a strategy pattern might work here? https://refactoring.guru/design-patterns/strategy/go/example